src/data/proxy/LocalTransientUserProfileStore.java

```diff
@@ -0,0 +1,67 @@
+package data.proxy;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Predicate;
+
+import data.proxy.utils.FunctionalUtils;
+import data.structure.UserProfile;
+
+/**
+ * LocalTransientUserProfileStore manages access to the stored user profiles locally in memory.
+ */
+public class LocalTransientUserProfileStore implements UserProfileStore {
```

Comment:
I initially had named it LocalUserProfileStore for that reason, but I realized that this may be a bit ambiguous with regards to the persistence or lack thereof of the user data. For example, it would not be obvious that it was not writing to files that would persist across JVM instances. I also didn't want to just name it TransientUserProfileStore, because then it wasn't obvious that it was completely isolated from any external systems.

If you feel strongly about it, then I can drop the "Transient" component, but otherwise, I'll keep it the way it is for the reasons given above.